maestro/tests/integration/test_deploy.py

```diff
@@ -0,0 +1,57 @@
+#!/usr/bin/env python3
+
+# Copyright © 2025 IBM
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import subprocess
+import requests
+
+import pytest
+from unittest import TestCase
+from src.deploy import Deploy
+
+# `deploy` tests
+class test_deploy(TestCase):
+    def setUp(self):        
+        self.cwd = os.getcwd()
+        os.chdir("src")
+        self.deploy = Deploy("../tests/examples/condition_agents.yaml", "../tests/examples/condition_workflow.yaml", "BEE_API_KEY=sk-proj-testkey BEE_API=http://192.168.86.45:4000 DRY_RUN=1")
+        #self.deploy = Deploy("../tests/examples/condition_agents.yaml", "../tests/examples/condition_workflow.yaml", "BEE_API_KEY=sk-proj-testkey BEE_API=http://192.168.86.45:4000")
+        
+    def tearDown(self):
+        self.deploy = None
+        os.chdir(self.cwd)
+
+    @pytest.mark.skipif(os.getenv('DEPLOY_KUBERNETES_TEST') != "1", reason="Kubernetes deploy skipped")
+    def test_deploy_to_kubernetes(self):
+        self.deploy.deploy_to_kubernetes()
+        #response = requests.get("http://127.0.0.1:30051/").text
```

Comment:
Ypiu may have to wait ... but definitely need to check the deployment. Thanks for that. I guess commented as you are working on it.